ffi/newpassmanagers.cpp

```diff
@@ -0,0 +1,319 @@
+}
+
+API_EXPORT(bool)
+LLVMPY_PTOGetLoopVectorization(LLVMPipelineTuningOptionsRef PTO) {
+    return llvm::unwrap(PTO)->LoopVectorization;
+}
+
+API_EXPORT(void)
+LLVMPY_PTOSetLoopVectorization(LLVMPipelineTuningOptionsRef PTO, bool value) {
+    llvm::unwrap(PTO)->LoopVectorization = value;
+}
+
+API_EXPORT(bool)
+LLVMPY_PTOGetSLPVectorization(LLVMPipelineTuningOptionsRef PTO) {
+    return llvm::unwrap(PTO)->SLPVectorization;
+}
+
+API_EXPORT(void)
+LLVMPY_PTOSetSLPVectorization(LLVMPipelineTuningOptionsRef PTO, bool value) {
+    llvm::unwrap(PTO)->SLPVectorization = value;
+}
+
+API_EXPORT(bool)
+LLVMPY_PTOGetLoopUnrolling(LLVMPipelineTuningOptionsRef PTO) {
+    return llvm::unwrap(PTO)->LoopUnrolling;
+}
+
+API_EXPORT(void)
+LLVMPY_PTOSetLoopUnrolling(LLVMPipelineTuningOptionsRef PTO, bool value) {
+    llvm::unwrap(PTO)->LoopUnrolling = value;
+}
+
+// FIXME: Available from llvm16
+// API_EXPORT(int)
+// LLVMPY_PTOGetInlinerThreshold(LLVMPipelineTuningOptionsRef PTO) {
+//     return llvm::unwrap(PTO)->InlinerThreshold;
+// }
+
+// API_EXPORT(void)
+// LLVMPY_PTOSetInlinerThreshold(LLVMPipelineTuningOptionsRef PTO, bool value) {
+//     llvm::unwrap(PTO)->InlinerThreshold = value;
+// }
+
+API_EXPORT(void)
+LLVMPY_DisposePipelineTuningOptions(LLVMPipelineTuningOptionsRef PTO) {
+    delete llvm::unwrap(PTO);
+}
+
+// PB
+
+API_EXPORT(LLVMPassBuilderRef)
+LLVMPY_CreatePassBuilder(LLVMTargetMachineRef TM,
+                         LLVMPipelineTuningOptionsRef PTO) {
+    TargetMachine *target = llvm::unwrap(TM);
+    PipelineTuningOptions *pt = llvm::unwrap(PTO);
+    return llvm::wrap(new PassBuilder(target, *pt));
+}
+
+API_EXPORT(void)
+LLVMPY_DisposePassBuilder(LLVMPassBuilderRef PB) { delete llvm::unwrap(PB); }
+
+static OptimizationLevel mapLevel(int speed_level, int size_level) {
+    switch (size_level) {
+    case 0:
+        switch (speed_level) {
+        case 0:
+            return OptimizationLevel::O0;
+        case 1:
+            return OptimizationLevel::O1;
+        case 2:
+            return OptimizationLevel::O2;
+        case 3:
+            return OptimizationLevel::O3;
+        default:
+            llvm_unreachable("Invalid optimization level");
+        }
+    case 1:
+        if (speed_level == 1)
+            return OptimizationLevel::Os;
+        llvm_unreachable("Invalid optimization level for size level 1");
+    case 2:
+        if (speed_level == 2)
+            return OptimizationLevel::Oz;
+        llvm_unreachable("Invalid optimization level for size level 2");
+    default:
+        llvm_unreachable("Invalid size level");
+        break;
+    }
+}
+
+API_EXPORT(LLVMModulePassManagerRef)
+LLVMPY_buildPerModuleDefaultPipeline(LLVMPassBuilderRef PBref, int speed_level,
+                                     int size_level) {
+
+    PassBuilder *PB = llvm::unwrap(PBref);
+    OptimizationLevel OL = mapLevel(speed_level, size_level);
+
+    // FXIME: No need to explicitly take care of O0 from LLVM 17
```

Comment:
Whilst I enjoy the irony of mis-spelling FIXME, I will sleep a little better at night knowing that it's not typoed:
```suggestion
    // FIXME: No need to explicitly take care of O0 from LLVM 17
```